integration-tests/slack/notify-circleci-test-results.js

```diff
@@ -0,0 +1,82 @@
+const _ = require('lodash/fp')
+const { getFailedTestNamesFromArtifacts } = require('./circleci-utils')
+const { getMessageBlockTemplate } = require('./message-templates')
+const { postMessage } = require('./post-message')
+const testsInfo = require('./slack-notify-channels.json')
+
+
+/**
+ * Get array of Slack channel IDs for succeeded job notification
+ * @returns {Array[string]}
+ */
+const getAllSlackChannelsForPassedJob = () => {
+  return _.map(_.get('id'), testsInfo.pass)
+}
+
+/**
+ * Get all Slack channel IDs and test names for failed job notification
+ * @returns {Map<string, Array[string]>}
+ */
+const getAllSlackChannelsForFailedJob = () => {
+  const allChannelsAndTests = new Map()
+  _.forEach(item => {
+    const channelId = _.get('id', item)
+    // const { tests } = item
+    const testNames = _.map(test => test.name, item.tests)
+    allChannelsAndTests.set(channelId, testNames)
+  }, testsInfo.fail)
+  return allChannelsAndTests
+}
+
+/**
+ * Get a map object of Slack channel IDs and array of test names for failed tests notification
+ * @param {Array[string]} failedTests
+ * @returns {Map<string, Array[string]>} A map object where key is channel_id, value is test_names array
+ */
+const getFailedTestsAndChannelIDs = failedTests => {
+  const allChannelsAndTests = getAllSlackChannelsForFailedJob()
+
+  const filteredIncludeOnlyFailedTests = new Map() // Map<string, Array[string]>
+  _.forEach(test => {
+    const channelIdsForFailedTest = Array.from(allChannelsAndTests.keys())
+      .filter(key => allChannelsAndTests.get(key).includes(test) || allChannelsAndTests.get(key).length === 0) // A channel without a list of tests means notify that channel for any failing test
+    if (channelIdsForFailedTest.length === 0) {
+      throw new Error(`Test: ${test} was not found in slack-notify-channels.json`)
+    }
```

Comment:
Since the current configuration always notifies the dsde-qa channel for any failing test, will this condition ever be true?

Reply:
Ah, I think we still need it. This check is good for ensuring JSON should also be updated when someone rename or add new test.
For example, Workspace team added a new test. No update to `slack-notify-channels.json`. When this test fails in future, `#dsde-qa` is the only channel that will receive Slack notification. Team that owns this test will never know until manually be told or update JSON in new PR.

Reply:
Oops, I'll check on this. You may be right about condition is never true.

Reply:
Sounds good. If we want to require all tests be configured to notify a specific Slack channel, it would be nice if that check happened before integration tests ran.